log4j-core/src/main/java/org/apache/logging/log4j/core/lookup/MapLookup.java

```diff
@@ -122,14 +122,14 @@ public String lookup(final LogEvent event, final String key) {
         if (map == null && !isMapMessage) {
             return null;
         }
-        if (map != null && map.containsKey(key)) {
-            final String obj = map.get(key);
+        if (isMapMessage) {
+            final String obj = ((MapMessage) event.getMessage()).get(key);
             if (obj != null) {
                 return obj;
             }
         }
-        if (isMapMessage) {
-            return ((MapMessage) event.getMessage()).get(key);
+        if (map != null && map.containsKey(key)) {
+            return map.get(key);
```

Comment:
`containsKey` immediately followed by `get` is unnecessary. If the map isn't null we can avoid the second traversal and return the result of `map.get(key)`.